### PR TITLE
(INFC-17763) Support Windows

### PR DESCRIPTION
**Pull request by danielparks**
This refactors the code and makes extensive changes to support both [native][] and Cygwin OpenSSH on Windows.

All that is required to release 1.0 is documentation, so this bumps the version to 0.9.0.

---

This has been extensively tested on 100+ nodes in the internal Puppet infrastructure.

[native]: https://github.com/PowerShell/openssh-portable

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -2,69 +2,204 @@
 #
 # Sets varables for the SSH class
 #
-class ssh::params {
```
Comment (danielparks) [inline manifests/params.pp:5]: This file is ridiculous. I decided not to pull it into this PR, but we should switch to hiera (https://github.com/ploperations/ploperations-ssh/issues/9).

File: metadata.json [hunk 2]
```diff
@@ -25,27 +45,34 @@
       ]
     },
     {
-      "operatingsystem": "RedHat",
```
Comment (danielparks) [inline metadata.json:28]: I sorted the operating systems alphabetically so I could compare this file to the one in ploperations-account.

File: files/sshd_linux [hunk 3]
```diff
@@ -1,12 +1,9 @@
-ChallengeResponseAuthentication yes
```
Comment (danielparks) [inline files/sshd_linux:1]: Sorted alphabetically.

File: metadata.json (hunk 2 above)
Comment (genebean) [reply metadata.json:28]: I like well sorted files better anyhow :)

File: manifests/params.pp [hunk 4]
```diff
@@ -2,69 +2,204 @@
...
+      $syslog_facility   = 'AUTHPRIV'
+      $sftp_subsystem    = 'internal-sftp'
+      $authorized_keys   = '.ssh/authorized_keys'
+      $strict_modes      = true
+      $print_motd        = true
+      $permit_root_login = false
+      $dos_line_endings  = false
     }
     'Solaris','SunOS': {
       case $::kernelrelease {
```
Comment (genebean) [inline manifests/params.pp]: Mind updating to the current structured fact here?

**Review by genebean — CHANGES_REQUESTED** [review]
Just one tiny change :)

File: manifests/params.pp (hunk 4 above)
Comment (danielparks) [reply manifests/params.pp]: Good call.

**Review by genebean — APPROVED** [review]

**Pull request merged**